README.md

```diff
@@ -33,15 +33,81 @@ val metadata: IO[RecordMetadata] = producer.use { producer =>
 ## Consumer usage example
 
 ```scala
+import cats.data.{NonEmptySet => Nes}
+
 val consumer = Consumer.of[IO, String, String](config, ecBlocking)
 val records: IO[ConsumerRecords[String, String]] = consumer.use { consumer => 
   for {
-    _       <- consumer.subscribe(Nel("topic"), None)
+    _       <- consumer.subscribe(Nes("topic"))
     records <- consumer.poll(100.millis)
   } yield records 
 }
 ```
 
+## Handling consumer group rebalance
+It's possible to provide a callback for a consumer group rebalance event, which can be useful if you want to do some computations,
+save the state, commit some offsets (or do anything with the consumer before the new group is applied).  
+This can be done by providing an implementation of `RebalanceListener1` (or a more convenient version - `RebalanceListener1WithConsumer`) 
+which requires you to return a `RebalanceCallback` structure which describes what actions should be performed in a certain situation. 
+It allows you to use some of consumer's methods as well as a way to run an arbitrary computation.
+Please note that all the actions are executed on the consumer `poll` thread which means that running heavy or 
+long-running computations is discouraged. 
+
+What you can currently do:
+- lift a pure value via `RebalanceCallback.pure(a)`
+- raise an error via `RebalanceCallback.fromTry(Failure(error))`
```

Comment:
I haven't looked at https://github.com/evolution-gaming/skafka/pull/146 yet, but mb we would be able to raise error via `ApplicativeError` after merging it?

Reply:
We surely would. But it would be a bit strange if this is merged before #146 and has a notion of functionality that isn't merged yet, no? If this is merged before #146 I could fix the doc in the corresponding PR

Reply:
sure, didn't mean to create any confusion, sorry :)
I'm ok with any of the following options
- merge #146 before this PR
- merge #146 after this PR, and mention error handling in the docs within #146
- merge #146 after this and reflect error handling in the docs via new dedicated PR
feel free to pick the one which is more convenient for you